config/auth_server.go

```diff
@@ -0,0 +1,161 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/databricks/databricks-sdk-go/logger"
+	"golang.org/x/oauth2"
+)
+
+// timeout to wait for local auth server to respond
+const authServerTimeout = 10 * time.Second
+
+type serverResponse struct {
+	// Optional URL of the Databricks host to connect to.
+	Host string `json:"host"`
```

Comment:
why not just use the `DATABRICKS_HOST` environment variable? it's a different thing from a token and is more "static", whereas a token is refreshable.

Reply:
I always pass a `HOST`, `TOKEN` pair so the host based auth can determine if the metadata service is applicable to the configured host.